File: Work/BaseORM/Kernel/Common/Sap.vb
Option Explicit On
Option Strict On

Imports SapORM.Contracts
Imports SapORM.Services


Namespace Kernel

    Public Class S

        Public Shared ReadOnly Property AP() As ISapDataService
            Get
                Dim sapDataService As ISapDataService

                If (Web.HttpContext.Current.Session("SapDataService") Is Nothing) Then
                    sapDataService = SapDataServiceFactory.Create()
                    Web.HttpContext.Current.Session("SapDataService") = sapDataService
                Else
                    sapDataService = CType(Web.HttpContext.Current.Session("SapDataService"), ISapDataService)
                End If

                Return sapDataService
            End Get
        End Property

        Private Shared _sapDataServiceFactory As ISapDataServiceFactory
        Public Shared ReadOnly Property SapDataServiceFactory() As ISapDataServiceFactory
            Get
                If (_sapDataServiceFactory Is Nothing) Then

                    _sapDataServiceFactory = New SapDataServiceDefaultFactory()

                End If

                Return _sapDataServiceFactory
            End Get
        End Property

    End Class

End Namespace

File: Work/Tools/CKGDatabaseAdminTool/CKGDatabaseAdminLib/Sap.cs
using SapORM.Contracts;
using SapORM.Services;

namespace CKGDatabaseAdminLib
{
    public class S
    {
        private static ISapDataService sapDataService;

        public static ISapDataService AP
        {
            get 
            { 
                if (sapDataService == null)
                {
                    sapDataService = SapDataServiceFactory.Create();
                }

                return sapDataService;
            }
        }

        public static ISapDataServiceFactory SapDataServiceFactory
        {
            get { return new SapDataServiceDefaultFactory(); }
        }
    }
}

